src/envoy/http/backend_routing/filter.cc

```diff
@@ -112,6 +112,9 @@ FilterHeadersStatus Filter::decodeHeaders(
       NOT_REACHED_GCOVR_EXCL_LINE;
   }
 
+  if (!headers.EnvoyOriginalPath()) {
+    headers.setEnvoyOriginalPath(headers.getPathValue());
```

Comment:
Not clear to me: why do we need this check? This is the first filter that runs before gRPC transcoding. Can't we just always override it?

Reply:
Nvm, this is wrong. Makes sense.